### ISSUE TITLE
Do port italics

### ISSUE THREAD

**Issue by J3Rik0** [PiotrGrochowski/Consolas#7]
`In my opinion the italic Consolas fonts of version 7.0 and before do more harm than good. It's better for italic to be handled by the rendering engine.`

Umm, no, just no, you may be a designer but I'm a really damn good tester and attention to details are backed up by my OCD in regards to the fine details between coding fonts, OS rendering engines and the editors themselves.

So, italics *SHOULD NEVER EVER* be left in the hands of the rendering engine, because the rendering engines will make the italic variant look oblique at up to 45 deg angle. I personally found that a discreet italic angle goes a really long way in giving a font a really stylistic completion feeling.

On top of all that, windows's rendering engine is f*cked up, the italics really look like they're at a 45 deg angle (see Fira Code that also doesn't have an italic variant), Linux is just a tad better, and OSX is probably the only one that doesn't exaggerate the italic angle but still to my eyes it just sucks. That's one of the reasons I won't and will never use Fira Code, when they'll add an italic I might reconsider.

So again, no, do put back the italics, if you won't work on them just put the old ones in there, for posterity at least.

I can really sit here and also touch on the difference between how java renders fonts vs the OS renderer and how some editors implement they own renderers and how a lot of designers have no clue about these differences and leave out the italics or pay no attention to the vertical alignment of -> and => or how to export the fonts with proper hinting that works on all platforms but I'll stop here.

I may not be a font designer but I believe I've tested every monospace font on the web, and while I appreciate the time and effort designers put into making a font and probably some of my rants are nitpicky I truly believe that a font aimed at programming should have an italic variant and also - or = should be properly vertically aligned with >. 

I for one am sticking with the default Consolas that comes with windows until this one also gets an italic variant, and I'm pretty sure a lot of programmers don't change to any other font is due to the beautiful italics it has, clean, elegant, and out of your way.

**Comment by PiotrGrochowski** [comment]
This isn't a font aimed just at programming. This is actually intended to [replace every font in existence](https://github.com/MicrosoftDocs/typography-issues/issues/183).

Certainly it doesn't look like "it is rendering at 45°". https://i.imgur.com/miVFirY.png

You should request how many degrees of slant you want and I can make a custom version for you. None of the characters will be modified otherwise or copied from Consolas Italic or Bold Italic 7.0. If you _like_ the strange characters of Consolas Italic and Bold Italic 7.0, I'm sorry, you'll have to deal with that horrible 2400 character coverage that's missing many characters.

**Comment by J3Rik0** [comment]
Yeah, I exaggerated a bit with the angle, but what you call perfect italics in your print isn't what I call perfect, it may be from a font designer's guidelines, but to my eyes it looks waaaay too slanted. I did a quick test and an angle of -6 is what, in my eyes, looks like a good balance between visually appealing and visually distinguishable for italic.

The books may say what a true italic should be, but for me, at least for a programming font, true italics need to be subtle, you need to be able to notice that it's slanted, but at the same time it needs to be a very small one, an elegant one if I may.

Here's my ("perfect") italic variant, with some automatic conversions from font forge to get a bit of its original italic appeal: [https://imgur.com/a/9lO5XB8](https://imgur.com/a/9lO5XB8)

Of course, my numbers are off, and that lowercase F needs to have its descender moved on the baseline.
It does also have some bugs, see the pic below, and the ascender serifs on the W, X and that K can be removed, the A and F in the first pic should suffice for a customization. [https://imgur.com/a/etf1Y3w](https://imgur.com/a/etf1Y3w)

**Comment by PiotrGrochowski** [comment]
What do you mean by -6?

What property of Consolas 8.x made you consider using it? Is it the few characters not in 7.0, the better hinting (only in the autohinted versions), the character modifications, something else?

Consolas 8.x is NOT developed specifically for programming. Therefore your "programming italics" will not be added to the repository. From my subjective point of view the way GDI renders italics looks much better than that 6° "programming" slant, although you have your own opinion.

I did your custom request though. The font is slanted with Element\Transformations\Transform, Skew..., not Element\Style\Italic... .

Consolas Italic 8.1: http://s000.tinyupload.com/index.php?file_id=05478757602472842603
Consolas Bold Italic 8.1: http://s000.tinyupload.com/index.php?file_id=19237666951477421483
